### PR TITLE
Feature/external share

### PR DESCRIPTION
**Pull request by Shahar-Y**
Added the external share component. Tested with docker-compose.

Need to approve with kubernetes after adding delegation-service to kd-helm.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ services:
     depends_on:
       - api-gateway
   api-gateway:
-    image: 'bluehub.azurecr.io/meateam/api-gateway:develop'
+    image: 'bluehub.azurecr.io/meateam/api-gateway:merges'
```
Comment (AllKinds) [inline docker-compose.yml]: change it back to develop

File: docker-compose.yml [hunk 2]
```diff
@@ -250,4 +249,57 @@ services:
...
+    environment:
+      PMTS_PORT: 8080
+      HOST_NAME: permit-service
+      PMTS_MONGO_HOST: mongodb://mongo:27017/permit
+      PMTS_GRANT_TYPE: client_credentials
+      PMTS_AUDIENCE: "_LCF5avgIpoXSWhY5Jl8mMi0Q~4_~R"
+      PMTS_APPROVAL_URL: http://23.102.32.203/api/v1/request
+      PMTS_SPIKE_SERVICE: spike-service:8080
+      PMTS_ELASTIC_APM_IGNORE_URLS: '/grpc.health.v1.Health/Check'
+      ELASTICSEARCH_URL: http://13.69.137.179:9200
```
Comment (yonatandt) [inline docker-compose.yml]: don't show IP

File: docker-compose.yml [hunk 3]
```diff
@@ -250,4 +249,57 @@ services:
...
+      LOG_INDEX: kdrive
+      LOG_LEVEL: debug
+      HEALTH_CHECK_INTERVAL: 5
+      ELASTIC_APM_SERVER_URL: 'http://13.69.137.179:8200'
```
Comment (yonatandt) [inline docker-compose.yml]: don't show IP

File: public/index.html [hunk 4]
```diff
@@ -84,17 +84,27 @@
       transform: scale(1.0);
     }
   }
+
+  img.omri-spinner {
```
Comment (yonatandt) [inline public/index.html]:
```suggestion
  img.loading-logo {
```

File: public/index.html [hunk 5]
```diff
@@ -84,17 +84,27 @@
...
 
   <div id="loading">
     <div class="spinner">
-      <div class="bounce1"></div>
+      <!-- <div class="bounce1"></div>
       <div class="bounce2"></div>
-      <div class="bounce3"></div>
+      <div class="bounce3"></div> -->
     </div>
+    <img src="/img/yesodot_logo.svg" alt="" class="omri-spinner">
```
Comment (yonatandt) [inline public/index.html]:
```suggestion
    <img src="/img/yesodot_logo.svg" alt="" class="loading-logo">
```

File: src/api/users.js [hunk 6]
```diff
@@ -12,6 +12,19 @@ export async function get (id) {
   return fetchJSON(`/api/users/${id}`, {})
 }
 
+export async function checkIfVIP (id) {
+  const vipRes = await axios.get(`${config.vipServiceUrl}/api/vips/${id}`);
+  return vipRes.data.vip === true;
+}
```
Comment (yonatandt) [inline src/api/users.js]: Not relevant anymore

File: src/store/getters.js [hunk 7]
```diff
@@ -9,6 +9,12 @@ const getters = {
   shares: state => state.shares,
   direction: state => state.user && state.user.locale === 'he' ? 'rtl' : 'ltr',
   isSearch: state => state.search,
+  getGlobalExternalUsers: state => state.globalExternalUsers,
+  getApprovers: state => state.approvers,
+  getStepThreeRes: state => state.stepThreeRes,
+  userID: state => state.user && state.user.id,
+  user: state => state.user,
+  isVIP: state => state.isVIP,
```
Comment (yonatandt) [inline src/store/getters.js]: Not relevant anymore

File: src/store/index.js [hunk 8]
```diff
@@ -7,6 +7,7 @@ Vue.use(Vuex)
 
 const state = {
   user: null,
+  isVIP: undefined,
```
Comment (yonatandt) [inline src/store/index.js]: Not relevant anymore

File: src/store/mutations.js [hunk 9]
```diff
@@ -42,6 +42,10 @@ const mutations = {
     i18n.default.locale = locale
     state.user = value
   },
+  setIsVIP: (state, value) => {
+    state.isVIP = value;
+    return;
+  },
```
Comment (yonatandt) [inline src/store/mutations.js]: Not relevant anymore

File: src/utils/constants.js [hunk 10]
```diff
@@ -25,9 +26,55 @@ const config = {
...
 	authUrl: '',
 	environment: '',
-	supportLink: ''
+	supportLink: '',
+	approvalServiceUrl: '',
+	externalShareName: 'שיתוף חיצוני',
+	myExternalSharesName: 'השיתופים החיצוניים שלי',
+	externalExclusiveUnit: 'יחידת ביג מאם',
+	enableExternalShare: false,
+	vipServiceUrl: 'localhost:8094',
```
Comment (yonatandt) [inline src/utils/constants.js]: Not relevant anymore

File: src/utils/constants.js [hunk 11]
```diff
@@ -55,6 +102,12 @@ export async function fetchConfig() {
 	config.authUrl = conf.authUrl;
 	config.environment = conf.environment;
 	config.supportLink = conf.supportLink;
+	config.approvalServiceUrl = conf.approvalServiceUrl || config.supportLink;
+	config.externalShareName = conf.externalShareName;
+	config.myExternalSharesName = conf.myExternalSharesName;
+	config.externalExclusiveUnit = conf.externalExclusiveUnit;
+	config.enableExternalShare = conf.enableExternalShare === "true";
+	config.vipServiceUrl = conf.vipServiceUrl;
```
Comment (yonatandt) [inline src/utils/constants.js]: Not relevant anymore

File: src/utils/constants.js [hunk 12]
```diff
@@ -55,6 +102,12 @@ export async function fetchConfig() {
 	config.authUrl = conf.authUrl;
 	config.environment = conf.environment;
 	config.supportLink = conf.supportLink;
+	config.approvalServiceUrl = conf.approvalServiceUrl || config.supportLink;
+	config.externalShareName = conf.externalShareName;
+	config.myExternalSharesName = conf.myExternalSharesName;
+	config.externalExclusiveUnit = conf.externalExclusiveUnit;
```
Comment (yonatandt) [inline src/utils/constants.js]: Not relevant anymore

File: src/utils/constants.js [hunk 13]
```diff
@@ -25,9 +26,55 @@ const config = {
 	apmServerUrl: '',
 	authUrl: '',
 	environment: '',
-	supportLink: ''
+	supportLink: '',
+	approvalServiceUrl: '',
+	externalShareName: 'שיתוף חיצוני',
+	myExternalSharesName: 'השיתופים החיצוניים שלי',
+	externalExclusiveUnit: 'יחידת ביג מאם',
```
Comment (yonatandt) [inline src/utils/constants.js]: Not relevant anymore

**Review by yonatandt — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (yonatandt) [reply docker-compose.yml]: Still here...

**Review by yonatandt — CHANGES_REQUESTED** [review]

**Comment by Shahar-Y** [comment]
Closing for #135

**Pull request closed without merging**